tests/components/smappee/test_config_flow.py

```diff
@@ -54,3 +273,153 @@ async def test_full_flow(hass, aiohttp_client, aioclient_mock):
 
     assert len(hass.config_entries.async_entries(DOMAIN)) == 1
     assert len(mock_setup.mock_calls) == 1
+
+
+async def test_full_zeroconf_flow(hass):
+    """Test the full zeroconf flow."""
+    with patch("pysmappee.api.SmappeeLocalApi.logon", return_value={}), patch(
+        "pysmappee.api.SmappeeLocalApi.load_advanced_config",
+        return_value=[{"key": "mdnsHostName", "value": "Smappee1006000212"}],
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_command_control_config", return_value=[]
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_instantaneous",
+        return_value=[{"key": "phase0ActivePower", "value": 0}],
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_ZEROCONF},
+            data={
+                "host": "1.2.3.4",
+                "port": 22,
+                CONF_HOSTNAME: "Smappee1006000212.local.",
+                "type": "_ssh._tcp.local.",
+                "name": "Smappee1006000212._ssh._tcp.local.",
+                "properties": {"_raw": {}},
+            },
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "zeroconf_confirm"
+        assert result["description_placeholders"] == {CONF_SERIALNUMBER: "1006000212"}
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "1.2.3.4"}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "Smappee1006000212"
+        assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+
+
+async def test_full_user_local_flow(hass):
+    """Test the full zeroconf flow."""
+    with patch("pysmappee.api.SmappeeLocalApi.logon", return_value={}), patch(
+        "pysmappee.api.SmappeeLocalApi.load_advanced_config",
+        return_value=[{"key": "mdnsHostName", "value": "Smappee1006000212"}],
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_command_control_config", return_value=[]
+    ), patch(
+        "pysmappee.api.SmappeeLocalApi.load_instantaneous",
+        return_value=[{"key": "phase0ActivePower", "value": 0}],
+    ):
```

Comment:
We should patch the `async_setup_entry` in the integration too, to avoid setting up the integration. Please do this for all create entry results.

Reply:
Why do we need to patch everything even async_init and async_configure methods? What's wrong with setting up the integration in a test env?

Reply:
Setting up the integration also does I/O and we might not patch that already. We normally patch entry `async_setup_entry` and assert that it's called.

Reply:
If you run the tests you'll notice that those two tests, the tests for the full flows, are blocking for many seconds during teardown.

Reply:
That's interesting! The block happens during the async_update method.